### PR TITLE
feat: Cluster node actions menu

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes will add an actions menu to the cluster node which will contain different actions like,
- Cordon node
- Drain node
- Edit taints
- Edit YAML
- Delete node

Fixes - [AB#1608](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1608)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually by visiting `Clusters > Select cluster > Perform node actions`.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ClusterNodes/NodeActions/NodeActionsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                            {nodeData.unschedulable ? 'Uncordon' : 'Cordon'}
+                        </span>
+                        <span
+                            className="flex left h-36 cursor pl-12 pr-12 dc__hover-n50"
+                            onClick={toggleShowDrainNodeDialog}
+                        >
+                            <DrainIcon className="mr-8" />
+                            Drain
+                        </span>
+                        <span className="flex left h-36 cursor pl-12 pr-12 dc__hover-n50" onClick={() => {}}>
```
Comment (vivek-devtron) [inline src/components/ClusterNodes/NodeActions/NodeActionsMenu.tsx:72]: We can use noop instead

File: src/components/ClusterNodes/NodeActions/NodeActionsMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                    <MenuDots className="node-actions-menu-icon icon-dim-16" />
+                </PopupMenu.Button>
+                <PopupMenu.Body>
+                    <div className="fs-13 fw-4 lh-20 pt-8 pb-8 w-160">
+                        <span
+                            className="flex left h-36 cursor pl-12 pr-12 dc__hover-n50"
+                            onClick={handleOpenTerminalAction}
+                        >
+                            <TerminalIcon className="mr-8" />
+                            Terminal
```
Comment (vivek-devtron) [inline src/components/ClusterNodes/NodeActions/NodeActionsMenu.tsx:56]: we should move these to const

File: src/components/ClusterNodes/types.ts [hunk 3]
```diff
@@ -149,3 +149,52 @@ export const TEXT_COLOR_CLASS = {
...
+}
+
+interface NodeActionRequest {
+    clusterId: number
+    name: string
+    version: string
+    kind: string
+}
+
+export interface NodeDeleteRequest extends NodeActionRequest {}
```
Comment (vivek-devtron) [inline src/components/ClusterNodes/types.ts:180]: We can use NodeActionRequest directly.

Comment (sohel-rp) [reply src/components/ClusterNodes/types.ts]: Done. Actually, I kept NodeDeleteRequest intentionally for any additional keys.

File: src/components/ClusterNodes/NodeActions/NodeActionsMenu.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/ClusterNodes/NodeActions/NodeActionsMenu.tsx]: Done. Moved all actions labels to const.

File: src/components/ClusterNodes/NodeActions/NodeActionsMenu.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/ClusterNodes/NodeActions/NodeActionsMenu.tsx:75]: It's intentional as it'll be changed with actual function in upcoming PR.

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**